File: TubeMapper/TubeMapper/Coordinator/AppCoordinator.m
//
//  AppCoordinator.m
//  TubeMapper
//
//  Created by Tim Searle on 18/11/2017.
//  Copyright © 2017 Tim Searle. All rights reserved.
//

#import "AppCoordinator.h"
#import "TubeCoordinator.h"

@implementation AppCoordinator

- (void)start {
    
    TubeCoordinator *tubeCoordinator = [[TubeCoordinator alloc] initWithNavigationController: self.navigationController];
    [self addChildCoordinator: tubeCoordinator];
    [tubeCoordinator start];    
}

@end

File: LastFM/LastFM/Coordinator/AppCoordinator.swift
//
//  AppCoordinator.swift
//  LastFM
//
//  Created by Tim Searle on 02/02/2018.
//  Copyright © 2018 Tim Searle. All rights reserved.
//

import Foundation
import Base

final class AppCoordinator: Coordinator {
    
    func start() {
        setupNetwork()
        
        let artistSearchCoordinator = ArtistCoordinator(navigationController: self.navigationController)
        
        self.add(child: artistSearchCoordinator)
        
        let artistSearchViewController = artistSearchCoordinator.searchViewController()
        self.navigationController.pushViewController(artistSearchViewController, animated: true)
    }
    
    private func setupNetwork() {
        Network.add(baseURL: BaseURL.lastfm.url().absoluteString)
    }
}
